awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,218 @@
+                'managed_by_policy': instance['managed_by_policy'],
+                }
+    return info
+
+
+@register('job_counts')
+def job_counts(since):
+    counts = {}
+    counts['total_jobs'] = models.UnifiedJob.objects.all().count()
+    counts['status'] = dict(models.UnifiedJob.objects.values_list('status').annotate(Count('status')))
+    counts['launch_type'] = dict(models.UnifiedJob.objects.values_list('launch_type').annotate(Count('launch_type')))
+    
+    return counts
+    
+    
+@register('job_instance_counts')
+def job_instance_counts(since):         #TODO: all of these are going to need to be restrained to the last 24 hours/INSIGHTS_SCHEDULE
+    counts = {}
+
+    job_types = models.UnifiedJob.objects.values_list(
+        'execution_node', 'launch_type').annotate(job_launch_type=Count('launch_type'))
+    for job in job_types:
+        counts.setdefault(job[0], {})[job[1]] = job[2]
+        
+    job_statuses = models.UnifiedJob.objects.values_list(
+        'execution_node', 'status').annotate(job_status=Count('status'))
+    for job in job_statuses:
+        counts.setdefault(job[0], {})[job[1]] = job[2]
+        
+# See if we can group by status and type inside the node grouping.  ^^
+
+    # 'jobs_running': models.UnifiedJob.objects.filter(execution_node=instance.hostname, status__in=('running', 'waiting',)).count(), # jobs in running & waiting state
+    # 'jobs_total': models.UnifiedJob.objects.filter(execution_node=instance.hostname).count(),
+    # ^^ try to get jobs_running and jobs_total per instance.  and uuid if easily possible  
+
+    return counts
+
+
+# Copies Job Events from db to a .csv to be shipped
+def copy_events_table(path='/tmp/'):
+    events_file = os.path.join(path, 'events_table.csv')
+    try:
+        write_data = open(events_file2, 'w')
+        with connection.cursor() as cursor:
+            cursor.copy_expert('''COPY (SELECT main_jobevent.id, 
+                                      main_jobevent.created, 
+                                      main_jobevent.modified, 
+                                      main_jobevent.event, 
+                                      main_jobevent.failed, 
+                                      main_jobevent.changed, 
+                                      main_jobevent.uuid, 
+                                      main_jobevent.playbook, 
+                                      main_jobevent.play, 
+                                      main_jobevent.role, 
+                                      main_jobevent.task, 
+                                      main_jobevent.counter, 
+                                      main_jobevent.verbosity, 
+                                      main_jobevent.start_line, 
+                                      main_jobevent.end_line, 
+                                      main_jobevent.job_id, 
+                                      main_jobevent.host_id, 
+                                      main_jobevent.host_name, 
```

Comment:
Same question re: `host_name`.  Do we _need_ this?

Reply:
If we identify problems connecting to certain hosts, we wouldn't be able to return which hosts were exhibiting issues.  But there is a strong argument against including host_name's from a PII perspective.  @ryanpetrello @wenottingham

Reply:
1. I don't see how you give information on flaky hosts without the host
2. Anyone using insights is already sending all their RHEL host information to RH

Reply:
ok, that makes sense, leaving the host_name in.